src/Discord.Net.Core/Entities/Users/IGuildUser.cs

```diff
@@ -85,6 +85,17 @@ public interface IGuildUser : IUser, IVoiceState
         /// </summary>
         int Hierarchy { get; }
 
+        /// <summary>
+        ///     Gets the date and time that indicates if and for how long a user has been timed out.
+        /// </summary>
+        /// <remarks>
+        ///     <see cref="null"/> or a timestamp in the past if the user is not timed out.
+        /// </remarks>
+        /// <returns>
+        ///     A <see cref="DateTimeOffset"/> indicating how long the user will be timed out for.
+        /// </returns>
+        DateTimeOffset? TimeoutDuration { get; }
```

Comment:
If the timestamp can be in the past, maybe we should add a getter here that checks if its in the past and returns null.

Reply:
Ontop of that, I didn't read the correctly yesterday. The name `TimedOutUntil` makes sense